src/mon/MonClient.cc

```diff
@@ -408,7 +408,9 @@ void MonClient::shutdown()
     delete version_requests.begin()->second;
     version_requests.erase(version_requests.begin());
   }
-
+  for (const auto cmd : mon_commands) {
+    _cancel_mon_command(cmd.first);
```

Comment:
this modifies mon_commands().  Does the magic for syntax advance the iterator before executing the body?  i'm guessing not, in which case you should do a while (!mon_commands.empty()) { auto& cmd = mon_commands.begin()->first; ...